File: taxe-38-backend11/src/main/java/org/bshg/taxe38/services/facade/Taxe38DetailService.java
package org.bshg.taxe38.services.facade;
import org.bshg.taxe38.entity.Taxe38Detail;
import org.bshg.taxe38.entity.Locale38Detail;
import org.bshg.taxe38.entity.Taxe38;
import org.bshg.taxe38.entity.TauxTaxe38;
import org.bshg.taxe38.zutils.pagination.Pagination;
import java.util.List;
public interface Taxe38DetailService {
Taxe38Detail findById(Long id);
List<Taxe38Detail> findAllOptimized();
List<Taxe38Detail> findAll();
Pagination<Taxe38Detail> findPaginated(int page, int size);
Taxe38Detail create(Taxe38Detail item);
List<Taxe38Detail> create(List<Taxe38Detail> item);
Taxe38Detail update(Taxe38Detail item);
List<Taxe38Detail> update(List<Taxe38Detail> item);
void deleteById(Long id);
void delete(Taxe38Detail item);
void delete(List<Taxe38Detail> items);
void deleteByIdIn(List<Long> ids);
int deleteByLocale38DetailId(Long id);
List<Taxe38Detail> findByLocale38DetailId(Long id);
int deleteByTauxTaxe38Id(Long id);
List<Taxe38Detail> findByTauxTaxe38Id(Long id);
int deleteByTaxe38Id(Long id);
List<Taxe38Detail> findByTaxe38Id(Long id);
}

File: taxe-38-frontend/src/app/controller/services/taxe-3-8-detail.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { environment } from 'src/environments/environment';
import {Pagination} from "src/app/controller/utils/pagination/pagination";
import { Taxe38Detail } from 'src/app/controller/entities/taxe-3-8-detail';
import { Taxe38DetailValidator } from 'src/app/controller/validators/taxe-3-8-detail.validator';

@Injectable({ providedIn: 'root' })
export class Taxe38DetailService {
  private readonly api = environment.apiUrl + "taxe38detail";
  private _item!: Taxe38Detail | undefined;
  private _items!: Array<Taxe38Detail>;
  private _pagination!: Pagination<Taxe38Detail>

  private http = inject(HttpClient)

  public keepData: boolean = false
  public returnUrl: string | undefined = undefined
  public toReturn = () => this.returnUrl != undefined


  public findAll() {
    return this.http.get<Array<Taxe38Detail>>(this.api);
  }

  public findById(id: number) {
    return this.http.get<Taxe38Detail>(`${this.api}/id/${id}`);
  }

  public findAllOptimized() {
    return this.http.get<Array<Taxe38Detail>>(`${this.api}/optimized`);
  }

  public findPaginated(page: number = 0, size: number = 10) {
    return this.http.get<Pagination<Taxe38Detail>>(`${this.api}/paginated?page=${page}&size=${size}`);
  }

  public create() {
    return this.http.post<Taxe38Detail>(this.api, this.item);
  }

  public createList() {
    return this.http.post<Array<Taxe38Detail>>(`${this.api}/all`, this.items);
  }

  public update() {
    return this.http.put<Taxe38Detail>(this.api, this.item);
  }

  public updateList() {
    return this.http.put<Array<Taxe38Detail>>(`${this.api}/all`, this.items);
  }

  public delete(dto: Taxe38Detail) {
    return this.http.delete<number>(this.api, {body: dto});
  }

  public deleteAll(dtos: Array<Taxe38Detail>) {
    return this.http.delete<number>(this.api, {body: dtos});
  }

  public deleteById(id: number) {
    return this.http.delete<number>(`${this.api}/id/${id}`);
  }

  public deleteByLocale38DetailId(id: number){
    return this.http.delete<number>(`${this.api}/locale38detail/${id}`);
  }

  public findByLocale38DetailId(id: number){
    return this.http.get<Array<Taxe38Detail>>(`${this.api}/locale38detail/${id}`);
  }
  public deleteByTauxTaxe38Id(id: number){
    return this.http.delete<number>(`${this.api}/tauxtaxe38/${id}`);
  }

  public findByTauxTaxe38Id(id: number){
    return this.http.get<Array<Taxe38Detail>>(`${this.api}/tauxtaxe38/${id}`);
  }
  public deleteByTaxe38Id(id: number){
    return this.http.delete<number>(`${this.api}/taxe38/${id}`);
  }

  public findByTaxe38Id(id: number){
    return this.http.get<Array<Taxe38Detail>>(`${this.api}/taxe38/${id}`);
  }

  //------------- getters and setters -----------------------
  public get itemIsNull(): boolean {
    return this._item == undefined
  }

  public get items() {
    if (this._items == undefined)
      this._items = [];
    return this._items;
  }

  get pagination() {
    if (this._pagination == null)
      this._pagination = new Pagination();
    return this._pagination;
  }

  set pagination(value) {
    this._pagination = value;
  }

  public set items(value) {
    this._items = value;
  }

  public get item(): Taxe38Detail {
    if (this._item == null)
      this._item = new Taxe38Detail();
    return this._item;
  }

  public set item(value: Taxe38Detail | undefined) {
    this._item = value;
  }

  public get createdItemAfterReturn() {
    let created = {
      item: this.item,
      created: this.toReturn()
    }
    this.returnUrl = undefined
    this.item = undefined
    return created
  }
}

